### ISSUE TITLE
147528092 Emendatation Search does not work when Soundex option is selected (Keith)

### ISSUE THREAD

**Issue by FreeUKGenIssues** [FreeUKGen/FreeUKRegProductIssues#1242]
Issue reported by **keithhayton** at 2019-09-04 12:01:32 UTC
Time: 2019-09-04T11:52:34+00:00
Session ID: 9746d281c498e162631e7f2f507b28f0
Problem Page URL: [/messages/communications](/messages/communications)
Previous Page URL: [https://www.freereg.org.uk/manage_resources/new?locale=en](https://www.freereg.org.uk/manage_resources/new?locale=en)
Reported Issue:
Searches for forenames using Soundex do not find some common abbreviations.  On the transcription for Claughton on Brock (Lancashire) St Thomas the Apostle (RC) church, there are numerous abbreviated forenames, as follows.
Search for Thomas using soundex does not find Th:
ditto for Robert does not find Rob:
ditto for Edward does not find Ed:
ditto for Margaret does not find Mar: - although search for Mary does incorrectly find this one
Search for Helen does not find Ellen (and vice versa)
ditto for Joseph nor for John does not find Jo: of Jo
ditto for Richard does not find Rich: or Rd
There may be others - this is just from one parish register transcript
    

![Screenshot](https://www.freereg.org.uk)
e0 u4 s9 C6:P19

**Comment by SteveBiggs** [comment]
Emendations should find these equivalent names. Is the colon on the end the problem?

**Comment by Captainkirkdawson** [comment]
Soundex was designed for surnames not forenames although it can be useful with forenames.

As Steve notes we normally find forename equivalences through emendations. 

With abbreviations we convert 3 termination punctuations  . fullstop, : colon and - hyphen. We do not convert : the semicolon

**Comment by Captainkirkdawson** [comment]
A code change would be simple. However correcting the existing database would be time consuming

**Comment by SteveBiggs** [comment]
It looks like these abbreviations use : colons (not  ; semicolons) so they should be found.

**Comment by richpomfret** [comment]
Needs further testing.

**Comment by SteveBiggs** [comment]
Have made some tests:

Search for Robert DOES find Rob: if Name Soundex is NOT ticked but -
if Soundex IS ticked Robert does NOT find Rob:

So selecting Soundex would appear to defeat emendations from working?

I cannot find the emendation tables, but it seems that two-letter abbreviations such as Th:, Ed:, etc. are not included. Should they be?

**Comment by Captainkirkdawson** [comment]
As noted above I would not have expected soundex to find similarities between these abbreviations and their expansions. It simply is not logical given the fundamental basis of soundex.
Most of the stated abbreviations are not in the emendation rules. Progress on this requires https://github.com/FreeUKGen/MyopicVicar/issues/823
It is NOT a bug

[edited to link to issue 823 PR]

**Comment by SteveBiggs** [comment]
I agree that soundex wouldn't find them, but I would expect emendations to work regardless of whether soundex is ticked.

Robert should return Rob: through emendations even if soundex is ticked shouldn't it?

**Comment by suffolkroots** [comment]
Search on Joseph does not return Jos'h, whether soundex on or off.

**Comment by stsccfr** [comment]
Soundex always ignores abbreviation marks, and crudely speaking, reduces a name to the first letter plus the next 3 consonantal sounds.  This means that if your search name is the conventional, full spelling, you are not likely to match any abbreviation, because the abbreviated form won't have as many consonantal sounds as the conventional spelling.  Thus Edward won't match Ed and Robert won't match Rob and Joseph won't match Josh, etc.

The original post here was about finding abbreviated forenames in particular.  Soundex is not the solution, but as Steve said, emendation rules are the way to handle abbreviations.  But emendation rules are very specific, and we can only afford to apply so many of them.  If we have an emendation rule linking Robert with "Rob." (with the full stop) then that is _all_ that we will match with that rule.  We will not match "Rob" or "Rob:" or "Rob;" or "Bob".  The old set of emendation rules didn't include any for the particular examples provided by the user.  However, the new set of rules implemented in FR #2779 covers nearly all of the ones mentioned in this case. They are based on frequency, so any not covered are so uncommon that they could not be included without exceeding the limit on the number of rules.
